docs/industrial_optimization/vignette_fss.rst

```diff
@@ -0,0 +1,149 @@
+Task completion times are encoded by continuous variables. 
+In the resulting formulation, the number of binary variables is quadratic in the number of jobs, 
+and the number of continuous variables is equal to the number of jobs multiplied by the number of tasks.
+
+The assigned constraints ensure that the jobs do not overlap, and the order of the jobs is respected across all machines.
+For each job, the number of constraints is quadratic in the number of machines.
+
+This formulation is used by the CQM solver, COIN-OR's Pulp CBC CMD solver, and SciPy's HiGHS.
+
+OR-Tools CP-SAT Solver 
+---------------------------
+
+OR-Tools defines a FSS formulation that fits their API where task processing times are passed as an :math:`n\times m` matrix.
+
+Nonlinear Solver Model 
+---------------------------
+
+D-Wave's NL solver can efficiently encode a FSS problem by taking advantage of a list variable that encodes
+an ordering of the jobs. This variable eliminates the need for both the quadratic number of binary variables 
+representing job orders and constraints preserving job order for the MILP formulation. The array of processing times
+is converted to a constant variable from which the task end times are computed. As a result, there is no need to 
+encode constraints that prevent the jobs from overlapping.
+
+The relevant Python code is as follows:
+
+.. testsetup:: 
+
+    processing_times = [[10, 5, 7], [20, 10, 15]]
+    num_jobs = 3
+    num_machines = 2
+
+>>> from dwave.optimization import Model
+>>>
+>>> model = Model()
+>>>
+>>> # Add the constant processing-times matrix 
+>>> times = model.constant(processing_times)
+>>>
+>>> # The decision symbol is a num_jobs-long array of integer variables 
+>>> order = model.list(num_jobs)
+>>> end_times = []
+>>>
+>>> for machine_m in range(num_machines):
+...     machine_m_times = []
+...     if machine_m == 0:
+...         for job_j in range(num_jobs):
+...             if job_j == 0:
+...                 machine_m_times.append(times[machine_m, :][order[job_j]])
+...             else:
+...                 end_job_j = times[machine_m, :][order[job_j]]
+...                 end_job_j += machine_m_times[-1]
+...                 machine_m_times.append(end_job_j)
+...     else:
+...         for job_j in range(num_jobs):
+...             if job_j == 0:
+...                 end_job_j = end_times[machine_m - 1][job_j]
+...                 end_job_j += times[machine_m, :][order[job_j]]
+...                 machine_m_times.append(end_job_j)
+...             else:
+...                 end_job_j = maximum(end_times[machine_m - 1][job_j], machine_m_times[-1])
+...                 end_job_j += times[machine_m, :][order[job_j]]
+...                 machine_m_times.append(end_job_j)
+...     end_times.append(machine_m_times)
+>>> makespan = end_times[-1][-1]
+>>> # The objective is to minimize the last end time
+>>> model.minimize(makespan)
+>>> model.lock()
```

Comment:
For the doctest failure ``Got:  <contextlib ...`` see other such examples, there are ways to either truncate the output shown or to prebent output